airflow/utils/task_group.py

```diff
@@ -401,6 +401,69 @@ def expand(self, arg: Iterable) -> "MappedTaskGroup":
             self.task_group._remove(self)
         return MappedTaskGroup(group_id=self._group_id, dag=self.dag, mapped_arg=arg)
 
+    def topological_sort(self, _include_subdag_tasks: bool = False):
+        """
+        Sorts children in topographical order, such that a task comes after any of its
+        upstream dependencies.
+
+        Heavily inspired by:
+        http://blog.jupo.org/2012/04/06/topological-sorting-acyclic-directed-graphs/
```

Comment:
Alas, this URL no longer works and is not in the Wayback Machine.

Reply:
Oh yeah :( I copied the doc over without checking it. I'll see if I can find a suitable replacement URL (and remove it from the doc string)

Reply:
Can't find what was on that URL, but I've worked out that this is an implementation of Kahn's Topo Sort algorithm.